org-jira.el

```diff
@@ -1758,20 +1758,10 @@ that should be bound to an issue."
 
 (defun org-jira-read-issue-type (&optional project)
   "Read issue type name.  PROJECT is the optional project key."
-  (let* ((issue-types
-          (mapcar 'cdr (if project
-                           (jiralib-get-issue-types-by-project project)
-                         (jiralib-get-issue-types))))
-         (initial-input (when (member (car org-jira-type-read-history) issue-types)
-                          org-jira-type-read-history)))
-    (completing-read
-     "Type: "
-     issue-types
-     nil
-     t
-     nil
-     initial-input
-     (car initial-input))))
```

Comment:
Meanwhile, this is what `completing-read` expects:
```
(completing-read PROMPT COLLECTION &optional PREDICATE REQUIRE-MATCH INITIAL-INPUT HIST DEF INHERIT-INPUT-METHOD)
```

Reply:
The arguments do look out of order, but looking at the prior call, do we still have `require-match` as `t`, and a working initial-input + hist  based on the completing-read defaults? (I can't check Emacs C-h at this moment)

The simplification looks to have just dropped all the `&optional` args entirely (vs re-aligning the argument order).

Reply:
I'll double-check. Yeah I dropped all the optional args just to confirm it worked; I'm happy to add some back.

Reply:
Actually, I think the variable names are just making this confusing - the arguments appear to be in order on Emacs 30.0.5 - this pattern is also used all over in org-jira:

```
(defun org-jira-read-priority ()
  "Read priority name."
  (completing-read
   "Priority: "
   (mapcar 'cdr (jiralib-get-priorities))
   nil
   t
   nil
   'org-jira-priority-read-history
   (car org-jira-priority-read-history)))
```

One such example is right before the block you updated.

I think we would want to fix it across the board, or scrutinize why this one is behaving differently and adjust it to work (vs wiping out the features of history, default history, and forced match - unless it's preferable to drop those everywhere).

I have a feeling the problem might be related to how "initial-input" is built (which is actually appearing like it should be the HIST) - I see in this area it's unquoted while in the other sample, the var for that slot is quoted :thinking:

Reply:
The error message in question was indeed complaining about the given arg not being a symbol.

Reply:
I think I messed this up 3 years ago when I undid adcdd53b461e11e22e5ff10035504f8f7535bf34 - my assumption at the time was the variable `initial-input` === the slot `INITIAL-INPUT` - since `HIST` does indeed accept a symbol or cons (not a list, as is used in INITIAL-INPUT).

I think we just need to restore the quote and perhaps for future self sake, rename the local variable to something indicating `hist` (like every other reference, that has `-history` as a suffix)

Reply:
Thanks for investigating this deeper!

Reply:
Thank you for bringing it to my attention!  I'm going to close this PR request, as I pushed the `'` fix with some notes for future refactoring.

Reply:
Ah no problem! Thanks for the fix!